server/repository/repository.proto

```diff
@@ -44,6 +44,28 @@ message RepoQuery {
 	string repo = 1;
 }
 
+// RepoAccessQuery is a query for checking access to a repo
+message RepoAccessQuery {
+	// The URL to the repo
+	string repo = 1;
+	// Username for accessing repo
+	string username = 2;
+	// Password for accessing repo
+	string password = 3;
+	// Private key data for accessing SSH repository
+	string PrivateKey = 4;
+	// Whether to skip certificate or host key validation
+	bool   Insecure = 5;
+}
+
+// RepoAccessResponse is the response to RepoAccessQuery
+message RepoAccessResponse {
+	// Whether access is possible
+	bool accessPossible = 1;
```

Comment:
Are these returned? I can't find mention elsewhere in the PR.

Reply:
Supposed to be used with the next commits - I think it's better to have a dedicated response rather than to display the raw error to the user, even if it'll contain basically the same message later on.